core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -286,6 +286,7 @@ class DAGScheduler(
       case None =>
         // We are going to register ancestor shuffle dependencies
         getAncestorShuffleDependencies(shuffleDep.rdd).foreach { dep =>
+          assert(!shuffleToMapStage.get(dep.shuffleId).isDefined)
```

Comment:
Yes, but it is not immediately obvious that that is inappropriate.  I need to spend some time re-familiarizing myself with newOrUsedShuffleStage.  In any event, just failing an assertion in the middle of the DAGScheduler is not likely something we want to do.  At a bare minimum, we'd want to be logging a more useful error message.